src/levanter/models/roberta.py

```diff
@@ -0,0 +1,852 @@
+        layer_norm = hnn.LayerNorm.init(axis=Embed, eps=config.layer_norm_eps)
+
+        decoder = hnn.Linear.init(Embed, Vocab, key=k_decoder, out_first=True)
+
+        # idk what this is: TODO
+        # self.bias = nn.Parameter(torch.zeros(config.vocab_size))
+        # self.decoder.bias = self.bias
+
+        return RobertaLMHead(dense, layer_norm, decoder)
+
+    @named_call
+    def __call__(self, features: NamedArray, *, key=None) -> NamedArray:
+        x = self.dense(features).rename({self.config.Embed: self.config.FinalEmbed})
+        x = hnn.gelu(x)
+        x = self.layer_norm(x)
+
+        # project back to size of vocabulary with bias
+        x = self.decoder(x)
+
+        return x
+
+class RobertaForMaskedLM(eqx.Module, StateDictSerializationMixin):
+    roberta: RobertaModel
+    lm_head: RobertaLMHead
+    Vocab: Axis
+
+    @classmethod
+    def init(self, Vocab: Axis, config: RobertaConfig, *, key):
+
+        # if config.is_decoder:
+        #     raise AttributeError("Model is being run as a MaskedLM aka an encoder model, but is_decoder is true")
+
+        k_rob, key_head = jrandom.split(key, 2)
+        roberta = RobertaModel.init(Vocab, config, add_pooling_layer=False, key=k_rob)
+        lm_head = RobertaLMHead.init(Vocab, config, key=key_head)
+
+        return RobertaForMaskedLM(roberta, lm_head, Vocab)
+
+    def get_output_embeddings(self):
+        return self.lm_head.decoder
+
+    def set_output_embeddings(self, new_embeddings):
+        self.lm_head.decoder = new_embeddings
+
+    def forward(
+        self,
+        input_ids: Optional[NamedArray] = None,
+        attention_mask: Optional[NamedArray] = None,
+        token_type_ids: Optional[NamedArray] = None,
+        position_ids: Optional[NamedArray] = None,
+        inputs_embeds: Optional[NamedArray] = None,
+        labels: Optional[NamedArray] = None,
+    ) -> Tuple[NamedArray]:
+        r"""
+        labels (`torch.LongTensor` of shape `(batch_size, sequence_length)`, *optional*):
+            Labels for computing the masked language modeling loss. Indices should be in `[-100, 0, ...,
+            config.vocab_size]` (see `input_ids` docstring) Tokens with indices set to `-100` are ignored (masked), the
+            loss is only computed for the tokens with labels in `[0, ..., config.vocab_size]`
+        kwargs (`Dict[str, any]`, optional, defaults to *{}*):
+            Used to hide legacy arguments that have been deprecated.
+        """
+
+        outputs = self.roberta(
+            input_ids,
+            attn_mask=attention_mask,
+            token_type_ids=token_type_ids,
+            position_ids=position_ids,
+            inputs_embeds=inputs_embeds
+        )
+
+        sequence_output = outputs[0]
+        prediction_scores = self.lm_head(sequence_output)
+
+        masked_lm_loss = None
+        if labels is not None:
+            masked_lm_loss = hnn.cross_entropy_loss(logits=prediction_scores, Label=self.Vocab, targets=labels)
```

Comment:
my preference is to put this part in a method called compute_loss that calls `__call__` (which computes the logits), and then computes the loss


also you'll want to use the loss mask when that gets merged here